### DISCUSSION TITLE
French translation group

### DISCUSSION THREAD

**Discussion by simplg** [microsoft/ML-For-Beginners#186] [Translations]
Welcome to the french translation group!

So translating becomes easier, we are hoping that you follow some directives about how we cooperate, how to review a translation and how to translate specific terms.

# How to cooperate
Before choosing any subject to translate, please check #166 for subjects being translated and this discussion.
When you've found an available subject, please tell others that you're doing it by replying to this discussion with the subject that you've chosen.
To start translating, fork this repository and translate in another branch.
When done, do a pull request with your translated files. You can ask for reviews of your translation by replying to this discussion.

# How to review
When reviewing, it is mandatory to check for misspelled words and Anglicism. Some words in English can't be kept as they are. If you are not sure how to translate something, please discuss about it here or in your PR, don't start an issue.

# How to translate 
Some terms appear multiple times along the courses. For consistency, it is therefore preferred that we agree to each other on a common vocabulary. If you encounter a word that isn't in this list and should be, please tell us about it!

| English | French |
| ------------- |:-------------:|
| Pre-lecture quiz | Quiz préalable |
| Post-lecture quiz | Quiz de validation des connaissances |
| Hidden patterns | Schémas non observés |
| Machine Learning | Machine Learning |
| Deep Learning | Deep Learning |
| Artificial intelligence | Intelligence artificielle |
| Neural networks | Réseaux neuronaux |
| Time Series | Série Chronologique |
| Natural Language Processing | Natural Language Processing |
| Fairlearn | Fairlearn |
| Fairness | Equité |
| Harms | Préjudices |
| Unfairness | Injustice |

## Links
Try to localize a maximum the external links whenever it's possible. For example, Wikipédia, Microsoft Documentation have usually a french localization. In the same idea, when a news article is linked, try to find a french newspaper which have also covered the same subject instead.
Internal links such as assignments or quizzes need to have their localization parameter added. Example:
* white-water-09ec41f0f.azurestaticapps.net/quiz/7/**?loc=fr**
* assignement.**fr**.md

We are happy to have you among us and hope you'll have fun translating!

**Comment by simplg** [comment]
A part les mots encore en discussion dans #183. J'aurai aussi une question pour "hidden patterns", je l'ai personnellement traduit en "schémas cachés" mais j'hésitais avec "schémas non observés" (comme utilisé ici https://www.quantmetry.com/blog/eacl-2021-resume-conferences/). Il pourrait être intéressant d'avoir vos opinions :)

**Reply by ManuSquall** [reply]
J'ai tenté de googliser les termes mais sans résultat. Je pense que garder le terme "schéma non observé" est mieux vu qu'un article y fait référence!

**Reply by simplg** [reply]
Ok, je modifierai ça et je l'ajoute dans la liste.
Pour la méthode de travail, j'ai une proposition: lorsqu'une personne veut se lancer dans la traduction d'un fichier, il le signale en premier dans la discussion et lorsque la PR est acceptée, on met à jour le ticket github avec le cours coché. Ca permet d'éviter que l'on traduise 2 fois le même fichier et de s'embêter avec un repo commun. Si vous avez une autre idée, je suis preneur !

**Reply by ManuSquall** [reply]
je pensais effectivement à quelque chose de similaire: signaler cela sous forme de message au niveau de l'issue de checklist mais le faire sur la discussion semble plus approprié.
Par contre peut-on éditer la checklist pour montrer une sorte d'état "en cours de traduction"? Par exemple (ou toute autre syntaxe):
![image](https://user-images.githubusercontent.com/29026887/125776321-ecee2bbe-3591-48f5-a4de-023fdd2a1251.png)

**Reply by simplg** [reply]
Il ne semble pas possible de le faire mais on pourrait ajouter un intitulé "en cours" entre parenthèses ? https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists#creating-task-lists

**Reply by ManuSquall** [reply]
Parfait ! Je peux rédiger une proposition dans ce sens au niveau de la partie "How to cooperate" plus haut

**Reply by simplg** [reply]
J'ai mis à jour la discussion avec certaines consignes, si vous voyez des choses à modifier ou ajouter, n'hésitez pas à me le signaler !

**Comment by ManuSquall** [comment]
Hello world, au cours de mes traductions j'ai eu à prendre l'habitude de traduire "Pre-lecture quiz" par "Quiz de pré-lecture" et "Post-lecture quiz" par "Quiz post-lecture" des fois. "Quiz de pré-conférence" et "Quiz de post-conférence" sont'ils plus adaptés selon vous?

**Reply by simplg** [reply]
Je ne suis malheureusement pas linguiste donc je ne saurai dire. Si vous avez traduit de cette manière, on peut continuer comme ça, je pense. Au mieux, on enlève le tiret afin de garder l'orthographe Canadienne ?
Selon le Figaro (et d'autres sites), on ne met pas de tiret après un préfixe comme "pré" : https://www.lefigaro.fr/langue-francaise/expressions-francaises/si-vous-connaissez-la-regle-du-trait-d-union-vous-etes-un-expert-de-la-langue-francaise-20190628 https://www.btb.termiumplus.gc.ca/tpv2guides/guides/clefsfp/index-fra.html?lang=fra&lettr=indx_catlog_p&page=9GNo5FFMUlOs.html

**Reply by ManuSquall** [reply]
Bon en même temps si on se fie à google translate 😶😶 :
![image](https://user-images.githubusercontent.com/29026887/125775071-c21fca3d-86fe-4301-af7c-0bb472dccfa8.png)

**Reply by simplg** [reply]
Je viens de regarder sur wiktionary et il semble que préconférence existe et se définit comme une animation se déoulant avant une conférence : https://fr.wiktionary.org/wiki/pr%C3%A9conf%C3%A9rence 
Donc on pourrait traduire par ça ? (sans le tiret vu que c'est la règle en français)
Cela semble être utilisé ici : https://www.ibo.org/fr/global-conference-the-hague/pre-conference/

**Reply by ManuSquall** [reply]
ok pour "préconférence", mais je ne trouve pas autant d'article par rapport à "postconférence" qui apparait parfois avec un tiret ou pas en français

**Reply by simplg** [reply]
Le parlement européen semble utiliser les deux... (https://eur-lex.europa.eu/LexUriServ/LexUriServ.do?uri=OJ:C:2004:103:0092:0095:FR:PDF https://www.europarl.europa.eu/sides/getDoc.do?pubRef=-//EP//TEXT+CRE+20040309+ITEMS+DOC+XML+V0//FR&amp;language=FR)
Peut être que les deux écritures sont possibles. On pourrait garder postconférence sans le tiret afin de rester consistant ?
Linguee semble avoir les deux comme résultat :
https://www.linguee.fr/francais-anglais/search?source=auto&query=post-conf%C3%A9rence
https://www.linguee.fr/francais-anglais/search?source=auto&query=postconf%C3%A9rence

**Reply by ManuSquall** [reply]
Effectivement:
![image](https://user-images.githubusercontent.com/29026887/125797126-fefcb7e4-7780-4f16-8961-969356ce5e4f.png)

**Reply by ManuSquall** [reply]
Faisons celà @simplg  et merci!

**Comment by ManuSquall** [comment]
@simplg  j'attends donc que tu assignes les valeurs au niveau du tableau de traduction plus haut pour corriger les anciennes traductions

**Comment by jlooper** [comment]
Ma foi, mon français se dégénère. Ma collègue m'a dit qu'en français on dit 'quizz' avec 2 'z'. Je vous laisse travailler, merci!!

**Reply by ManuSquall** [reply]
I don't think so 😅, you're right. C'est bel et bien "quiz".
Ref : https://zero-faute-d-orthographe.com/quiz-ou-quizz , https://www.zcorrecteurs.fr/forum/sujet-1515-quiz-ou-quizz.html

**Comment by maud-lv** [comment]
Bonjour ! Je suis la collègue de Jen, on peut dire "quizz" ou "quiz" en français. Personnellement j'ai l'habitude d'écrire quizz avec deux "z" mais ça dépend des habitudes de chacun.
Je suggère "Quiz préalable" pour "pre-lecture quiz" et "Quiz de validation des connaissances" pour "post-lecture quiz". Je vous laisse voir si vous préférez écrire "quiz" ou "quizz".

**Reply by ManuSquall** [reply]
Je pense aussi que ces traductions ("Quiz préalable" et "Quiz de validation des connaissances") donnent beaucoup plus le sens voulu. @simplg ?

**Reply by ManuSquall** [reply]
@simplg I can't edit your first mmessage, can you add the translation of "machine learning" which still stay as "machine learning" as discussed here : https://github.com/microsoft/ML-For-Beginners/pull/167#issuecomment-879356952

Do we keep "deep learning" as it is also?
What do you think about it? I'm on the quiz app translation

**Reply by simplg** [reply]
No problem, I'm going to edit the first message. Concerning pre-lecture and post-lecture, do we keep it as it is or change it to the suggestion above? So I know if I change it here and in my PR

**Reply by simplg** [reply]
I think it is better to keep them in English for 3 reasons:
1. A lot of literature and media in french use them like that.
2. Ressources are often in English, so knowing the original terms might be better for further reading and comprehension.
3. Sometimes, some words translated to French can be "weird" (personal feeling)

However, it might be useful at the first mention of them to translate it between parentheses. So people know their equivalent and our Canadian friends won't try to punch us 😆

**Reply by jlooper** [reply]
@ManuSquall about quizzes - once translated, all the links to the quizzes can change to include a parameter: ?loc=fr

**Reply by ManuSquall** [reply]
we can manage that. 
All the links localized in readme.fr.md and assignment.fr.md files right? @jlooper

**Comment by ManuSquall** [comment]
Hello world, about "Time Series" which translate is better : "séries chronologiques" or "séries temporelles". I have seen both on differents articles.

**Reply by jlooper** [reply]
qu'en pensez-vous: https://docs.microsoft.com/fr-Fr/learn/modules/explore-analyze-time-series-insights/

**Reply by ManuSquall** [reply]
Effectivement, on peut aussi le laisser tel quel à l'image du Deep Learning et Machine Learning cité plus haut! Merci!

**Reply by simplg** [reply]
https://docs.microsoft.com/fr-Fr/learn/modules/explore-analyze-time-series-insights/1-introduction Ils ont traduit par "Série chronologique".
Azure Time Series est un nom de produit donc il n'est pas traduit par contre.

**Comment by simplg** [comment]
Je suis en train de traduire "02 - History of ML". Je suis tombé sur NLP (Natural Language Processing) qui semble se traduire par TALN (traitement automatique du langage naturel) en français:
* https://fr.wikipedia.org/wiki/Traitement_automatique_des_langues
* https://www.ionos.fr/digitalguide/web-marketing/vendre-sur-internet/le-traitement-automatique-du-langage-naturel-taln/
* https://datascientest.com/introduction-au-nlp-natural-language-processing

Vous en pensez quoi ?

**Reply by ManuSquall** [reply]
Hello @simplg, au cours de la traduction du quizz, c'est les initiales NLP que j'ai continué à utiliser, même pour la définition je l'ai conservé en anglais dans le doute

**Reply by simplg** [reply]
Ok, on le garde en anglais donc ? Dans ce cas je mets juste l'équivalent français entre parenthèse ?

**Reply by ManuSquall** [reply]
Yup, on peut faire ca, merci!

**Comment by ManuSquall** [comment]
Hello world, j'espère que vous vous portez bien.
Dans un soucis de traduction du module 1-3-fairness assignment je me demandais si conserver le terme "Fairlearn" n'était pas plus adéquat à l'image de ce qui a été fait dans cet [article](https://ichi.pro/fr/les-prejuges-comptent-qu-est-ce-que-fairlearn-et-pourquoi-devrais-je-m-en-soucier-130684373648844)?

**Reply by jlooper** [reply]
Fairlearn est une marque, on devrait conserver le mot tel quel 🙏

**Reply by simplg** [reply]
For the same module, I've added the translation for **fairness**, **harms** and **unfairness** to the list.

**Comment by simplg** [comment]
I'm currently translating the last module for part 1 (techniques of ml).
